common/OpTestHost.py

```diff
@@ -1180,6 +1180,110 @@ def host_build_fwts_tool(self, i_dir):
             raise OpTestError(l_msg)
 
     ##
+    # @brief Check that host has a CAPI FPGA card
+    #
+    # @return BMC_CONST.FW_SUCCESS or raise OpTestError
+    #
+    def host_has_capi_fpga_card(self):
+        l_cmd = "lspci -d \"1014:0477\""
+        print l_cmd
+        l_res = self.host_run_command(l_cmd)
+        print l_res
+        if (l_res.__contains__('IBM Device 0477')):
+            l_msg = "Host has a CAPI FPGA card"
+            print l_msg
+        else:
+            l_msg = "Host has no CAPI FPGA card"
+            print l_msg
+            raise OpTestError(l_msg)
```

Comment:
All the systems may not be having capi card, instead of raising the error, just print a message says like "No CAPI card, skipping the tests..." and just return it without running the tests(i.e skip the tests if there is no CAPI card). or you can catch this OpTestError exception in your four tests and return there if test got this exception.